app/models/bot/alegre.rb

```diff
@@ -5,6 +5,37 @@ class Bot::Alegre < BotUser
     attr_accessor :alegre_similarity_thresholds
   end
 
+  DynamicAnnotation::Field.class_eval do
+    after_save :save_analysis_to_similarity_index, if: :can_be_sent_to_index?
+    after_destroy :delete_analysis_from_similarity_index, if: :can_be_sent_to_index?
+    
+    def self.save_analysis_to_similarity_index(pm_id)
+      pm = ProjectMedia.find(pm_id)
```

Comment:
It's better to use `find_by_id` instead of `find` to avoid through error for non existing media

Reply:
Fair point! I'll change it.